chainer/functions/connection/convolution_2d.py

```diff
@@ -201,17 +220,43 @@ def backward_gpu(self, inputs, grad_outputs):
                     algo, workspace.data.ptr, workspace_size,
                     zero.data, self.filter_desc.value, gW.data.ptr)
 
-                if not self.deterministic:
-                    algo = libcudnn.getConvolutionBackwardDataAlgorithm(
-                        handle, self.filter_desc.value, gy_desc.value,
-                        self.conv_desc.value, x_desc.value, _bwd_data_pref,
-                        workspace_size)
+                conv_desc_bd = self.conv_desc
+                if hasattr(self, 'conv_desc_fp32'):
+                    # WA for cuDNN 6.0 or below.
+                    # When stride > 1, fp16 is slower than fp32.
+                    # Try to remove the WA, when new cuDNN is released.
+                    if self.sx != 1 or self.sy != 1:
+                        conv_desc_bd = self.conv_desc_fp32
+
+                    try:
+                        # try to find algo for fp16 compute
+                        algo = libcudnn.getConvolutionBackwardDataAlgorithm(
+                            handle, self.filter_desc.value, gy_desc.value,
+                            conv_desc_bd.value, x_desc.value, _bwd_data_pref,
+                            workspace_size)
+                    except RuntimeError:
+                        # print("No algorithm found for fp16 compute (kh:{}, kw:{}, n:{}, c:{}, h:{}, w:{}, out_c:{}, out_h:{}, out_w:{})".format(kh, kw, n, c, h, w, out_c, out_h, out_w))  # NOQA
```

Comment:
I feel it would be better to show the warning, but most users may not like to see the warning like this. So, I will leave to you as to whether or not to show the warning.